controllers/repo_manager/controller.go

```diff
@@ -148,14 +148,26 @@ func (r *RepoManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	// If ingress_type==ingress the operator should fail in case no ingress_class provided
-	// To avoid errors with clusters configured without or with multiple default IngressClass we will ask users to pass an ingress_class
 	// in case of ingress_type == ingress.
 	if isIngress(pulp) {
+
+		// If ingress_type==ingress the operator should fail in case no ingress_class provided
+		// To avoid errors with clusters configured without or with multiple default IngressClass we will ask users to pass an ingress_class
 		if len(pulp.Spec.IngressClassName) == 0 {
 			log.Error(nil, "ingress_type defined as ingress but no ingress_class_name provided.")
 			return ctrl.Result{}, fmt.Errorf("please, define the ingress_class_name field (with the name of the IngressClass that the operator should use to deploy the new Ingress) to avoid unexpected errors with multiple controllers available")
 		}
+
+		// the operator should also fail in case no ingress_host is provided
+		// ingress_host is used to populate CONTENT_ORIGIN and ANSIBLE_API_HOSTNAME vars from settings.py
+		// https://docs.pulpproject.org/pulpcore/configuration/settings.html#content-origin
+		//   "A required string containing the protocol, fqdn, and port where the content app is reachable by users.
+		//   This is used by pulpcore and various plugins when referring users to the content app."
+		if len(pulp.Spec.IngressHost) == 0 {
+			log.Error(nil, "ingress_type defined as ingress but no ingress_host provided.")
+			return ctrl.Result{}, fmt.Errorf("please, define the ingress_host field with the fqdn where Pulp should be accessed. This field is required to access API and also redirect Pulp CONTENT requests")
+		}
```

Comment:
please use deployment type instead of pulp

Reply:
my bad :/